### PR TITLE
Use entity.sensors().* etc, instead of deprecated methods

### PR DESCRIPTION
**Pull request by aledsage**

### CODE REVIEW

File: core/src/test/java/org/apache/brooklyn/core/entity/PolicyRegistrationTest.java [hunk 1]
```diff
@@ -79,7 +80,7 @@ public void setUp() throws Exception {
     
     @Test
     public void testGetPoliciesIsInitiallyEmpty() {
-        assertEquals(entity.getPolicies(), ImmutableList.of());
+        assertEquals(ImmutableList.copyOf(entity.policies()), ImmutableList.of());
```
Comment (alasdairhodge) [inline core/src/test/java/org/apache/brooklyn/core/entity/PolicyRegistrationTest.java]: Maybe suggest Guava's `Iterables.elementsEqual()` here (and elsewhere) but no strong feelings.

File: core/src/test/java/org/apache/brooklyn/entity/group/DynamicGroupTest.java [hunk 2]
```diff
@@ -476,18 +473,24 @@ public void testDoesNotDeadlockOnUnmanageWhileOtherMemberBeingAdded() throws Exc
...
+                            throw Exceptions.propagate(e);
+                        }
                     }
+                    super.emit(sensor, val);
                 }
-                super.emit(sensor, val);
+            };
+            @Override
+            public BasicSensorSupport sensors() {
+                return interceptedSensors;
```
Comment (alasdairhodge) [inline core/src/test/java/org/apache/brooklyn/entity/group/DynamicGroupTest.java]: Good catch!

**Comment by alasdairhodge** [comment]
Lovely jubbly: goodbye, compiler warnings! One teeny-tiny suggestion, but good to merge as it.

**Comment by aledsage** [comment]
Thanks; merging now and will address comments in separate PR.

**Pull request merged**